app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,19 @@
+class CommentsController < ApplicationController
+  def index
+    puts
+    redirect_to "/users/#{params['user_id']}/posts/#{params['post_id']}"
+  end
+
+  def new
+    @comment = self
+  end
+
+  def create
+    @values = params[:comment]
+    @comment = Comment.new(text: @values['text'], author: User.find_by(id: params['user_id']),
+                           post: Post.find_by(id: params['post_id']))
+    return unless @comment.save
+
+    redirect_to "/users/#{params['user_id']}/posts/#{params['post_id']}"
+  end
```

Comment:
# Optional:
I will suggest you add the params method. Open the [strong parameters](https://guides.rubyonrails.org/action_controller_overview.html#strong-parameters) method here to learn more
- Example
```ruby
 private 
  def comment_params
    params.require(:comment).permit(:text)
  end
```